lib/apiservers/portlayer/restapi/handlers/containers_handlers.go

```diff
@@ -255,6 +256,13 @@ func (handler *ContainersHandlersImpl) RemoveContainerHandler(params containers.
 		case exec.RemovePowerError:
 			return containers.NewContainerRemoveConflict().WithPayload(&models.Error{Message: err.Error()})
 		default:
+			if f, ok := err.(types.HasFault); ok {
+				switch f.Fault().(type) {
+				case *types.HostNotConnected:
+					p := &models.Error{Message: "Couldn't remove container. The ESX host is temporarily disconnected. Please try again later."}
+					return containers.NewContainerRemoveDefault(http.StatusInternalServerError).WithPayload(p)
```

Comment:
You should use NewContainerRemoveInternalServerError() hear instead of RemoveDefault.

Reply:
right right